README.md

```diff
@@ -1,2 +1,214 @@
+* **The Runtime Utility**: this is a collection of helper functions and defined constants that are used throughout Runtime. Most of them have to do with certain Runtime configuration values, maximum sizes for certain messages, or retrieving information about the different types of `lowcar` devices.
+
+This README will not go into each of these parts into exhaustive detail; explanations for each part can be found in the corresponding folder in the repo for the interested reader. However, we will describe briefly the data that flows between the various parts, and the manner in which that data is sent:
+
+* `net_handler` communicates with both Dawn and Shepherd. It receives start and stop commands, input data for the coding challenge, and information about the gamepad state. It sends log messages, data about the connected devices, and output data for the coding challenge. This data is sent over both TCP and UDP connections (depending on the type of data) to Shepherd and Dawn, and the data is packaged using Google Protocol Buffers.
+* `dev_handler` and `net_handler` are connected via `shm_wrapper`. This connection is used for the `net_handler` to fetch the most recent state of the connected devices, which is then sent to Dawn for the students to see.
+* `executor` and `net_handler` are connected via `shm_wrapper_aux` and a FIFO pipe. The connections are used to pass information about the coding challenge (both inputs to the functions and the student's outputs), and for `executor` to know what to run at any given time (autonomous mode, teleop mode, coding challenges, or idle).
+* `executor` and `dev_handler` are connected via `shm_wrapper`. This connection is used for `executor` to send commands to the attached devices (tell a motor to run at a certain speed, move a servo to certain position, etc.), and for `dev_handler` to serve `executor` with the device data that it needs to run the student code.
+* `dev_handler` communicates with `lowcar` via serial connection. This connection is used to poll for new devices, detect when devices have disconnected, and send device data and commands between the Raspberry Pi and the Arduinos.
+* All three processes output log messages through the `logger` tool. If logs are to be sent over the network to Dawn, those logs are put into a FIFO pipe which is opened by `net_handler`, where the logs are processed and sent to Dawn.
+
+## Concepts
+
+Runtime uses a **lot** of systems concepts, which is hopefully why you're here--to learn about systems! The following is a list of some various concepts that you'll be exposed to in Runtime, along with where they are used within Runtime. Keep in mind, this is _not_ a list of prerequisite knowledge you need to have in order to work on Runtime!
+
+* **Networking**
+    * Used to connect Runtime to Dawn and Shepherd
+    * **Unreliable Datagram Protocol (UDP)**
```

Comment:
Not sure if you meant this as a joke but it really stands for User Datagram Protocal hahahaha.

Reply:
Oh dammit yes it does idk what I was thinking here, it was not a joke in my head 😂